node/__tests__/request_coordinator.test.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable node/no-unpublished-import */
+
+const originalDns = jest.requireActual('dns');
+
+const markedResolver = {
+  resolveSrv: jest.fn(() => Promise.resolve([{name: 'host', port: 0}])),
+  resolve4: jest.fn(() => Promise.resolve(['ip1', 'ip2'])),
+};
+
+jest.mock('dns', () => ({
+  ...originalDns,
+  promises: {
+    Resolver: jest.fn(() => markedResolver),
+  },
+}));
+
+const markedAdminClient = {
+  pause: jest.fn(() => Promise.resolve()),
+  unpause: jest.fn(() => Promise.resolve()),
+  checkPaused: jest.fn(() => Promise.resolve(false)),
+  getHost: jest.fn(),
+};
+
+jest.mock('../src/admin_client', () => ({
+  AdminClient: jest.fn(() => markedAdminClient),
+}));
+
+import {RequestCoordinator} from '../src/request_coordinator';
+
+beforeEach(() => {
+  markedAdminClient.pause.mockRestore();
+  markedAdminClient.unpause.mockRestore();
+});
+
+test('pause successfully', async () => {
+  const coordinator = new RequestCoordinator('srv');
+
+  await coordinator.pause(false);
+
+  expect(markedAdminClient.pause).toBeCalledWith(false, null);
+  expect(markedAdminClient.pause).toBeCalledTimes(2);
```

Comment:
I've approved it, but it would be better to check if the pause requests are going to separate IPs like ip1 and ip2 in addition to checking if there are 2 requests.
Ditto for other test cases.

Reply:
Thank you for your advice.

I think this can be more precise https://github.com/scalar-labs/scalar-admin/pull/23/commits/244190bfbbd32a54c0857ea0b29dc419e879adbb

Please take a look again!